spyder/app/mainwindow.py

```diff
@@ -2375,40 +2376,48 @@ def report_issue(self, traceback=""):
         revision = ''
         if versions['revision']:
             revision = versions['revision']
+        reminder_message = (
+            "**PLEASE READ:** Before reporting here, *please* carefully "
+            "read our **Troubleshooting Guide**: <{0!s}> and search the issue "
+            "page for your error/problem, as most "
+            "posted bugs are duplicates or easy fixes.\n\n"
+            "Also, if you do post here, please describe step by step in "
+            "detail what you were doing when the error occurred. "
+            "Otherwise, your issue will likely be closed after 7 days. "
+            "Thanks.\n\n"
+            "**Note:** Delete this message before submitting."

```

Comment:
Does this part go to the text published on Github when users press the *Submit to Github* button in our report errors dialog?

Reply:
Yes, so the apparent majority of users who are sending auto-generated error reports, or error reports from the Help menu, which appears to be the dominant source of "drive by" traceback only reports (rather than navigating to Github and hitting the "New Issue" button, where that text is filled from the .github issue template) will see it _before_ they actually hit the submit their user-side/duplicate/incomplete issue and can get help faster. Or, if they don't remove it, we know they didn't even bother to read at it (as if they did, they would have seen the note saying it should be removed), giving stronger cause to summarily close it if desired.

Therein lies the problem, as mentioned above—due to the issue discussed in #5247 , at least in Firefox 52 ESR error reports are getting cut off around or below (due to % encoding) ~1300 real characters, and after editing it and the error template itself way down, perhaps more than you're comfortable with, that leaves around 100 characters of buffer after the dependencies are included. This should be okay for now for reports from the help menu, as they don't ever include more characters than this, and there's room for 2 or 3 more deps if need be, and to refine the format a bit more (maybe adding back a few words/headings here and there). Also, the change to the actual issue template is a non-issue since its done from Github, not Spyder.

However, this is a yuge problem for the autogenerated reports in many browsers, because it only leaves room for the shortest of tracebacks, and anything beyond that starts cutting off the dependencies, or (if long enough) the traceback itself. While this is already a problem (see #5247 ), this would exacerbate it.

Therefore, I suggest for now:

* Updating the Github issue template with the blurb, (currently implemented with this PR) as it is not affected
* Slimming down the existing basic structure of the Spyder-sent report where possible (currently implemented, and I can apply a few more tweaks to save a few more chars), to help a little more with traceback cutoff for now
* Not including the disclaimer (or maybe a one-line one for now) if a traceback is included, to leave room for the latter (will be implemented in this PR momentarily), otherwise, including it for now (implemented here)
* Including a interstitial dialog displaying the blurb if "Send to Github" is pressed from the error dialog, as originally proposed (future PR), or even better just integrating the text with the dialog from #5346 (that other PR, or direct followup)
* Some better alternative to the current "send the full text of the template in a GET request through the browser" method, that either just sends the traceback and the dep versions, maybe encoded in a char saving format, and then adds the static parts later...somehow...or use the Github API, or send the former to a Snyder-team-controlled webpage that adds the latter, prompts the user for the description, and sends that to Github by a non-truncated means...but maybe that's complicating things. (In the future, and probably not by me, but hopefully soon-ish)

In any case, I will submit a new commit soon with the immediate changes mentioned above.

Reply:
> Yes

Ok, so this needs PR #5346 to be merged first because @rlaverde discovered that many tracebacks are incomplete because it seems people are sending reports with Internet Explorer, which has a limit of 2000 chars in its URL bar.

> Including a interstitial dialog displaying the blurb if "Send to Github"

Good idea. Please get the branch in PR #5346 and open a new PR with it and your new changes.

> Some better alternative to the current "send the full text of the template in a GET request through the browser" method

Our idea is to force users to paste the contents of the error contained in the dialog into Github, instead of sending the whole error to the browser.

Reply:
> Ok, so this needs PR #5346 to be merged first because @rlaverde discovered that many tracebacks are incomplete because it seems people are sending reports with Internet Explorer, which has a limit of 2000 chars in its URL bar.

Well, the original PR yes, eventually to accomplish all the above bullets, but not as it now stands. As discussed at (perhaps too much) length in my previous comment, I've revised what I have here to take that into account, so it is a flat improvement on the current situation for the time being:

* Adds the full error blurb to the semi-automated and manual Github reports with enough chars   to spare for 3 additional dependencies even on the dev version (with a longer Spyder version code + SHA1), 
* Has 46 more chars for tracebacks (vs. the current ``3.x``) with a one-liner version of the blurb
* Includes an additional data point (OS Version) and refined formatting)
* Also helps mitigate the cut-off problem by a simple implementation of

> Our idea is to force users to paste the contents of the error contained in the dialog into Github, instead of sending the whole error to the browser

And, crucially for getting merged promptly (ideally, for ``3.2.6``), it accomplishes this this without actually changing any existing UI text or program functionality, just the text/formatting of the error reports.

As mentioned, from my testing it appears this is also a problem on (at least) Firefox 52 ESR and Chrome on Windows 8.1 (which should natively support it); so it likely happens with **all browsers on Windows**. It seems _many_ reports are cut off right now, especially the dependencies. Therefore, a `3.2.6` merge on this would help mitigate this until `3.2.7` or `3.2.8`, if that's still possible.

Reply:
Please don't write so long comments on the revision thread but on the comments of the PR itself.

Reply:
Reasons for me to ask you that:

1. Your comments have very little to do with my simple question.
2. Now I have to skip this long thread to do other revisions.

Reply:
I'm sorry; I was unaware of that convention. Would you like me to delete them and repost them on the main thread?